### PR TITLE
docs(replays): Publish recording segment endpoints

### PR DESCRIPTION
**Pull request by cmanallen**
Related: https://github.com/getsentry/team-replay/issues/353

### CODE REVIEW

File: src/sentry/replays/endpoints/project_replay_recording_segment_details.py [hunk 1]
```diff
@@ -5,13 +5,18 @@
...
 from sentry.api.api_owners import ApiOwner
 from sentry.api.api_publish_status import ApiPublishStatus
 from sentry.api.base import region_silo_endpoint
 from sentry.api.bases.project import ProjectEndpoint
+from sentry.apidocs.constants import RESPONSE_BAD_REQUEST, RESPONSE_FORBIDDEN, RESPONSE_NOT_FOUND
+from sentry.apidocs.examples.replay_examples import ReplayExamples
+from sentry.apidocs.parameters import GlobalParams
+from sentry.apidocs.utils import inline_sentry_response_serializer
 from sentry.replays.lib.storage import RecordingSegmentStorageMeta, make_filename
 from sentry.replays.usecases.reader import download_segment, fetch_segment_metadata
```
Comment (michellewzhang) [inline src/sentry/replays/endpoints/project_replay_recording_segment_details.py:22]: should we add
```suggestion
@extend_schema(tags=["Replays"])
```
?

Comment (cmanallen) [reply src/sentry/replays/endpoints/project_replay_recording_segment_details.py:22]: Yes we should.  Nice catch :D

File: src/sentry/replays/endpoints/project_replay_recording_segment_index.py [hunk 2]
```diff
@@ -10,6 +11,10 @@
 from sentry.api.base import region_silo_endpoint
 from sentry.api.bases.project import ProjectEndpoint
 from sentry.api.paginator import GenericOffsetPaginator
+from sentry.apidocs.constants import RESPONSE_BAD_REQUEST, RESPONSE_FORBIDDEN, RESPONSE_NOT_FOUND
+from sentry.apidocs.examples.replay_examples import ReplayExamples
+from sentry.apidocs.parameters import CursorQueryParam, GlobalParams
+from sentry.apidocs.utils import inline_sentry_response_serializer
 from sentry.replays.lib.storage import storage
 from sentry.replays.usecases.reader import download_segments, fetch_segments_metadata
```
Comment (michellewzhang) [inline src/sentry/replays/endpoints/project_replay_recording_segment_index.py:20]: similarly here too?
```suggestion
@extend_schema(tags=["Replays"])
```

File: src/sentry/replays/endpoints/project_replay_recording_segment_index.py [hunk 3]
```diff
@@ -18,13 +23,24 @@
...
+        "GET": ApiPublishStatus.PUBLIC,
     }
 
     def __init__(self, **options) -> None:
         storage.initialize_client()
         super().__init__(**options)
 
+    @extend_schema(
+        operation_id="List the recording segments for a replay",
+        parameters=[CursorQueryParam, GlobalParams.ORG_SLUG, GlobalParams.PROJECT_SLUG],
```
Comment (michellewzhang) [inline src/sentry/replays/endpoints/project_replay_recording_segment_index.py:35]: think we need a `PER_PAGE` one here (but i'm going off of the blueprint)

File: src/sentry/replays/endpoints/project_replay_recording_segment_details.py [hunk 4]
```diff
@@ -5,25 +5,43 @@
...
+    @extend_schema(
+        operation_id="Retrieve a replay recording segment",
+        parameters=[GlobalParams.ORG_SLUG, GlobalParams.PROJECT_SLUG],
+        responses={
+            200: inline_sentry_response_serializer("ListReplayRecordingSegments", list[dict]),
+            400: RESPONSE_BAD_REQUEST,
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOT_FOUND,
+        },
+        examples=ReplayExamples.GET_REPLAY_SEGMENTS,
```
Comment (michellewzhang) [inline src/sentry/replays/endpoints/project_replay_recording_segment_details.py:41]:
```suggestion
        examples=ReplayExamples.GET_REPLAY_SEGMENT,
```

File: src/sentry/replays/endpoints/project_replay_recording_segment_details.py [hunk 5]
```diff
@@ -5,25 +5,43 @@
...
     publish_status = {
-        "GET": ApiPublishStatus.UNKNOWN,
+        "GET": ApiPublishStatus.PUBLIC,
     }
 
+    @extend_schema(
+        operation_id="Retrieve a replay recording segment",
+        parameters=[GlobalParams.ORG_SLUG, GlobalParams.PROJECT_SLUG],
+        responses={
+            200: inline_sentry_response_serializer("ListReplayRecordingSegments", list[dict]),
```
Comment (michellewzhang) [inline src/sentry/replays/endpoints/project_replay_recording_segment_details.py:36]: would be great to type the `list[dict]`!

File: src/sentry/replays/endpoints/project_replay_recording_segment_index.py [hunk 6]
```diff
@@ -10,22 +11,39 @@
...
+        responses={
+            200: inline_sentry_response_serializer("ListReplayRecordingSegments", list[list[dict]]),
```
Comment (michellewzhang) [inline src/sentry/replays/endpoints/project_replay_recording_segment_index.py:38]: the type here could then just be a `list` of the other type you defined
```suggestion
            200: inline_sentry_response_serializer("ListReplayRecordingSegments", list[ReplayRecordingSegmentResponse]),
```

File: src/sentry/replays/endpoints/project_replay_recording_segment_details.py [hunk 7]
```diff
@@ -5,25 +5,43 @@
...
+@extend_schema(tags=["Replays"])
 class ProjectReplayRecordingSegmentDetailsEndpoint(ProjectEndpoint):
     owner = ApiOwner.REPLAY
     publish_status = {
-        "GET": ApiPublishStatus.UNKNOWN,
+        "GET": ApiPublishStatus.PUBLIC,
     }
 
+    @extend_schema(
+        operation_id="Retrieve a replay recording segment",
```
Comment (michellewzhang) [inline src/sentry/replays/endpoints/project_replay_recording_segment_details.py:33]:
```suggestion
        operation_id="Retrieve a Replay Recording Segment",
```

File: src/sentry/replays/endpoints/project_replay_recording_segment_index.py [hunk 8]
```diff
@@ -10,22 +11,39 @@
...
-        "GET": ApiPublishStatus.UNKNOWN,
+        "GET": ApiPublishStatus.PUBLIC,
     }
 
     def __init__(self, **options) -> None:
         storage.initialize_client()
         super().__init__(**options)
 
+    @extend_schema(
+        operation_id="List the recording segments for a replay",
```
Comment (michellewzhang) [inline src/sentry/replays/endpoints/project_replay_recording_segment_index.py:35]:
```suggestion
        operation_id="List the Recording Segments for a Replay",
```

File: src/sentry/replays/endpoints/project_replay_recording_segment_details.py (hunk 5 above)
Comment (cmanallen) [reply src/sentry/replays/endpoints/project_replay_recording_segment_details.py:36]: Too big!  Its RRWeb so I think I'm going to leave it untyped for now.

**Review by michellewzhang — APPROVED** [review]
gif

**Comment by michellewzhang** [comment]
seeing a failing test for `List Recording Segments` -- i think we need `REPLAY_ID` param for that one

File: src/sentry/replays/types.py [hunk 9]
```diff
@@ -0,0 +1 @@
+ReplayRecordingSegment = list[dict]
```
Comment (schew2381) [inline src/sentry/replays/types.py:1]: Is the dict in this list the same as:
https://github.com/getsentry/sentry/blob/7a00c559245d9a7fc4cd944723753434aad53155/src/sentry/replays/usecases/ingest/__init__.py#L27-L29

I'm just guessing here but it seems like the details endpoint is fetching a`Recording Segment` which itself is a list, and index is fetching is a list of `Recording Segments` which ends up being a list of lists

Comment (cmanallen) [reply src/sentry/replays/types.py:1]: No unfortunately not.  ReplayRecordingSegment in this context is RRWeb which is a very large and complicated specification.  I don't think its necessary or possible to encapsulates its specification with a TypedDict.

**Pull request merged**